### PR TITLE
Prac 07 feedback

### PR DESCRIPTION
**Pull request by jessepurcell**
Hi @bigdogmitchyp
Could you please look at the code readabilty and docstrings. I'd also like you to take a close look at the get_valid_number method and the update_project method.

### CODE REVIEW

File: prac_07/guitar.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        return f"{self.name} ({self.year}) : ${self.cost:.2f}"
+
+    def __repr__(self):
+        return f"({self.__str__()})"
+
+    def get_age(self):
+        return 2023 - self.year
+
+    def is_vintage(self):
+        return self.get_age() >= 50
```
Comment (bigdogmitchyp) [inline prac_07/guitar.py:17]: could use docstrings here perhaps

File: prac_07/myguitars.py
Comment (bigdogmitchyp) [inline prac_07/myguitars.py:1]: perfect i reckon

File: prac_07/language_file_reader.py
Comment (bigdogmitchyp) [inline prac_07/language_file_reader.py:1]: looks great

File: prac_07/programming_language.py
Comment (bigdogmitchyp) [inline prac_07/programming_language.py:1]: great stuff

File: prac_07/project_management.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+def get_valid_string(display_string):
+    """Gets a valid string from the user."""
+    user_input = input(display_string)
+    while user_input == "":
+        print("Input can not be blank.")
+        user_input = input(display_string)
+    return user_input
+
+
+def get_valid_number(display_string):
```
Comment (bigdogmitchyp) [inline prac_07/project_management.py:135]: looks good

File: prac_07/project_management.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    new_percentage_string = input("New percentage: ")
+    if new_percentage_string:
+        while not is_valid_number:
+            try:
+                new_percentage = int(new_percentage_string)
+                project_choice.completion_percentage = new_percentage
+                is_valid_number = True
+            except TypeError:
+                print("Invalid number")
+                new_percentage_string = input("New percentage: ")
```
Comment (bigdogmitchyp) [inline prac_07/project_management.py:81]: looks familar to the get_valid_number function, maybe could have been used

File: prac_07/project_management.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    new_priority_string = input("New priority: ")
+    if new_priority_string:
+        while not is_valid_number:
+            try:
+                new_priority = int(new_priority_string)
+                project_choice.priority = new_priority
+                is_valid_number = True
+            except TypeError:
+                print("Invalid number")
+                new_priority_string = input("Priority: ")
```
Comment (bigdogmitchyp) [inline prac_07/project_management.py:94]: could be a function like the code above

File: prac_07/project_management.py
Comment (bigdogmitchyp) [inline prac_07/project_management.py:1]: codes readability is great and makes sense

File: prac_07/guitar.py (hunk 1 above)
Comment (jessepurcell) [reply prac_07/guitar.py:17]: Agreed

**Pull request merged**